### DISCUSSION TITLE
Github Copilot support?

### DISCUSSION THREAD

**Discussion by nomike** [TheR1D/shell_gpt#365] [Ideas]
I have a free account for Chat-GPT wjhich doesn't seem to allow me to use shell-gpt. But I have a paid subscription for Github Copilot which also uses a language model from OpenAI as it's backend.

As far as I can tell, shell-gpt doesn't support it as a backend yet, but IMHO this would help a lot of people.

Is there someone working on this already?

I was trying to find an official API documentation for Copilot but all I've found so far is the vscode plugin. Has someone fond a link to the API already? If I have that, I might give it a shot to implement support for Copilot.

**Comment by TheR1D** [comment]
GitHub Copilot does not have a public API. There are no plans to integrate Copilot into ShellGPT until GitHub announces official public API. Thank you for your interest in the project. Closing this issue due to its age and lack of similar requests from other users.

**Reply by Justincept** [reply]
microsoft 365 key code
